File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        stack = []

        dummy = ListNode(0)
        curr = dummy
        
        while head:
            stack.append(ListNode(head.val))
            head = head.next
        
        while stack:
            curr.next = stack.pop()
            curr = curr.next
        
        return dummy.next
            

File: 0206-reverse-linked-list/0206-reverse-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        Stack<ListNode> stack = new Stack<ListNode>();

        ListNode dummy = new ListNode(0);
        ListNode curr = dummy;

        while (head != null) {
            stack.Push(new ListNode(head.val));
            head = head.next;
        }

        while (stack.Count > 0){
            curr.next = stack.Pop();
            curr = curr.next;
        }

        return dummy.next;
    }
}